automata/memory_store/symbol_doc_embedding.py

```diff
@@ -68,7 +70,8 @@ def _update_existing_embedding(
     ) -> None:
         existing_embedding = self.embedding_db.get(symbol.full_dotpath)
         if (
-            existing_embedding.symbol != symbol
+            self.overwrite
```

Comment:
This still needs fixing, overwrite should be moved up to where we select which to update.